grr/server/grr_response_server/bin/grrafana.py

```diff
@@ -51,22 +64,128 @@ class JSONResponse(werkzeug_wrappers_json.JSONMixin,
+    records_list = fleetspeak_utils.FetchClientResourceUsageRecords(
+        client_id, start_range_ts, end_range_ts)
+    record_values = self.record_values_extract_fn(records_list)
+    datapoints = [(v, r.server_timestamp.seconds * 1000 +
+                   r.server_timestamp.nanos // 1000000)
+                  for (v, r) in zip(record_values, records_list)]
+    return _TargetWithDatapoints(target=self.name, datapoints=datapoints)
+
+
+class ClientsStatisticsMetric(Metric):
+
+  def __init__(self, name: str, get_fleet_stats_fn: Callable,
+               days_active: int) -> None:
+    super().__init__(name)
+    self.get_fleet_stats_fn = get_fleet_stats_fn
+    self.days_active = days_active
+
+  def ProcessQuery(self, req: JSONRequest) -> _TableQueryResult:
+    fleet_stats = self.get_fleet_stats_fn(_FLEET_BREAKDOWN_DAY_BUCKETS)
+    totals = fleet_stats.GetTotalsForDay(self.days_active)
+    return _TableQueryResult(
+      columns=[{
+        "text": "Label", "type":"string"
+      }, {
+        "text": "Value", "type": "number"
+      }],
+      rows=[[l, v] for l, v in totals.items()],
+      type="table")
+
+
+AVAILABLE_METRICS_LIST: List[Metric]
+AVAILABLE_METRICS_LIST = [
+    ClientResourceUsageMetric(
+        "Mean User CPU Rate", lambda records_list:
+        [record.mean_user_cpu_rate for record in records_list]),
+    ClientResourceUsageMetric(
+        "Max User CPU Rate", lambda records_list:
+        [record.max_user_cpu_rate for record in records_list]),
+    ClientResourceUsageMetric(
+        "Mean System CPU Rate", lambda records_list:
+        [record.mean_system_cpu_rate for record in records_list]),
+    ClientResourceUsageMetric(
+        "Max System CPU Rate", lambda records_list:
+        [record.max_system_cpu_rate for record in records_list]),
+    # Converting MiB to MB
+    ClientResourceUsageMetric(
+        "Mean Resident Memory MB", lambda records_list:
+        [record.mean_resident_memory_mib * 1.049 for record in records_list]),
+    ClientResourceUsageMetric(
+        "Max Resident Memory MB", lambda records_list:
+        [record.max_resident_memory_mib * 1.049 for record in records_list]),
+]
+AVAILABLE_METRICS_LIST.extend([
+    ClientsStatisticsMetric(
+        f"OS Platform Breakdown - {n_days} Day Active",
+        lambda buckets: data_store.REL_DB.CountClientPlatformsByLabel(buckets),
+        n_days) for n_days in _FLEET_BREAKDOWN_DAY_BUCKETS
+])
+AVAILABLE_METRICS_LIST.extend([
+    ClientsStatisticsMetric(
+        f"OS Release Version Breakdown - {n_days} Day Active", lambda buckets:
+        data_store.REL_DB.CountClientPlatformReleasesByLabel(buckets), n_days)
+    for n_days in _FLEET_BREAKDOWN_DAY_BUCKETS
+])
+AVAILABLE_METRICS_LIST.extend([
+    ClientsStatisticsMetric(
+        f"Client Version Strings - {n_days} Day Active", lambda buckets:
+        data_store.REL_DB.CountClientVersionStringsByLabel(buckets), n_days)
+    for n_days in _FLEET_BREAKDOWN_DAY_BUCKETS
+])
```

Comment:
I think these cram too much stuff into single expressions and are hard to parse for human. Rather than using `extend` with list comprehensions, try to write this with classic `for` loop and `append`s.